services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/map/ProcessedMapRepositoryImpl.java

```diff
@@ -0,0 +1,289 @@
+            }
+        }
+
+        List<MapBoundingBox> mapsContainingPoints = index.mapsContainingPoint(new CoordinateXY(longitude, latitude));
+        
+        List<IntersectionReferenceData> result = new ArrayList<>();
+        for(MapBoundingBox box: mapsContainingPoints){
+            ProcessedMap<LineString> map = mapLookup.get(box.getIntersectionId());
+            IntersectionReferenceData data = new IntersectionReferenceData();
+            data.setIntersectionID(map.getProperties().getIntersectionId());
+            data.setRoadRegulatorID("-1");
+            data.setRsuIP(map.getProperties().getOriginIp());
+
+            if(map.getProperties().getIntersectionName() != null && map.getProperties().getIntersectionName().isEmpty()){
+                data.setIntersectionName(map.getProperties().getIntersectionName());
+            }
+            
+            if (map.getProperties().getRefPoint() != null) {
+                data.setLatitude(map.getProperties().getRefPoint().getLatitude().doubleValue());
+                data.setLongitude(map.getProperties().getRefPoint().getLongitude().doubleValue());
+            }
+            result.add(data);
+        }
+
+        return result;
+
+    }
+
+    public List<IDCount> getMapBroadcastRates(int intersectionID, Long startTime, Long endTime) {
+
+        String startTimeString = Instant.ofEpochMilli(0).toString();
+        String endTimeString = Instant.now().toString();
+
+        if (startTime != null) {
+            startTimeString = Instant.ofEpochMilli(startTime).toString();
+        }
+        if (endTime != null) {
+            endTimeString = Instant.ofEpochMilli(endTime).toString();
+        }
+        AggregationOptions options = AggregationOptions.builder().allowDiskUse(true).build();
+
+        Aggregation aggregation = Aggregation.newAggregation(
+                Aggregation.match(Criteria.where("properties.intersectionId").is(intersectionID)),
+                Aggregation.match(Criteria.where("properties.timeStamp").gte(startTimeString).lte(endTimeString)),
+                Aggregation.project("properties.timeStamp"),
+                Aggregation.project()
+                        .and(DateOperators.DateFromString.fromStringOf("timeStamp")).as("date"),
+                Aggregation.project()
+                        .and(DateOperators.DateToString.dateOf("date").toString("%Y-%m-%d-%H")).as("dateStr"),
+                Aggregation.group("dateStr").count().as("count"),
+                Aggregation.sort(Sort.Direction.ASC, "_id")).withOptions(options);
+
+        AggregationResults<IDCount> result = mongoTemplate.aggregate(aggregation, collectionName, IDCount.class);
+        List<IDCount> results = result.getMappedResults();
+
+        for (IDCount r : results) {
+            r.setCount((double) r.getCount() / 3600.0);
+        }
+
+        return results;
+    }
+
+    public List<IDCount> getMapBroadcastRateDistribution(int intersectionID, Long startTime, Long endTime) {
```

Comment:
suggestion: XML style comments on these methods to explain what they are doing. Might also be good to add comments in the method itself to outline the algorithms we're following.

Reply:
Added javadoc and method comments

Reply:
This is a great start with the commenting, let's add a backlog item to cover the rest of the functions as well

Reply:
Created new backlog item to cover this: https://dev.azure.com/SOC-OIT/CDOT/_backlogs/backlog/CDOT%20Connected%20Vehicles/Backlog%20items/?workitem=290540